arviz/plots/bpvplot.py

```diff
@@ -0,0 +1,294 @@
+"""Bayesian p-value Posterior/Prior predictive plot."""
+import numpy as np
+from matplotlib.colors import to_hex
+
+from .plot_utils import (
+    xarray_var_iter,
+    _scale_fig_size,
+    default_grid,
+    filter_plotters_list,
+    get_plotting_function,
+)
+from ..rcparams import rcParams
+from ..utils import _var_names
+
+
+def plot_bpv(
+    data,
+    kind="u_value",
+    t_stat="median",
+    bpv=True,
+    plot_mean=True,
+    reference="samples",
+    n_ref=100,
+    hdi_prob=0.94,
+    color="C0",
+    figsize=None,
+    textsize=None,
+    data_pairs=None,
+    var_names=None,
+    filter_vars=None,
+    coords=None,
+    flatten=None,
+    flatten_pp=None,
+    ax=None,
+    backend=None,
+    plot_ref_kwargs=None,
+    backend_kwargs=None,
+    group="posterior",
+    show=None,
+):
+    """
+    Plot Bayesian p-value for observed data and Posterior/Prior predictive.
+
+    Parameters
+    ----------
+    data : az.InferenceData object
+        InferenceData object containing the observed and posterior/prior predictive data.
+    kind : str
+        Type of plot to display ("p_value", "u_value", "t_stat"). Defaults to u_value.
+        For "p_value" we compute p := p(y* ≤ y | y). This is the probability of the data y being
+        larger or equal than the predicted data y*. The ideal value is 0.5 (half the predictions
+        below and half above the data).
+        For "u_value" we compute pi := p(yi* ≤ yi | y). i.e. like a p_value but per observation yi.
+        This is also known as marginal p_value. The ideal distribution is uniform. This is similar
+        to the LOO-pit calculation/plot, the difference is than in LOO-pit plot we compute
+        pi = p(yi* r ≤ yi | y-i ), where y-i, is all other data except yi.
+        For "t_stat" we compute := p(T(y)* ≤ T(y) | y) where T is any T statistic. See t_stat
+        argument below for details of available options.
+    t_stat : str, float, or callable
```

Comment:
How about adding a `"identity"` option too to encourage users to calculate T statistics manually and store them in idata object and then use these new variables as variable names with `t_stat="identity"`? 

I know it is already possible to do using `t_stat=lambda x: x`, hence the _encourage_ above. This would be helpful if T statistic function was expensive to calculate or for convenience to use T statistic with both `plot_bpv` and `loo_pit`